src/flb_network.c

```diff
@@ -577,28 +577,35 @@ static int flb_net_ares_sock_create_callback(ares_socket_t socket_fd,
 
     event_mask = MK_EVENT_READ;
 
-    /* c-ares doesn't use a macro for the socket type so :
+    /*
+     * c-ares doesn't use a macro for the socket type so :
      * 1 means it's a TCP socket
      * 2 means it's a UDP socket
      *
      * For TCP sockets we want to monitor for write events because we need to call
      * ares_process_fd in order to issue the query unlike UDP sockets which automatically
      * send the query after creating the socket.
      */
-    if(1 == type){
+    if (1 == type) {
         event_mask |= MK_EVENT_WRITE;
     }
 
-    mk_event_add(context->event_loop, socket_fd, FLB_ENGINE_EV_CUSTOM,
-                 event_mask, &context->response_event);
+    ret = mk_event_add(context->event_loop, socket_fd, FLB_ENGINE_EV_CUSTOM,
+                       event_mask, &context->response_event);
+    if (ret != 0) {
+        return -1;
+    }
 
     return ARES_SUCCESS;
 }
 
-struct flb_dns_lookup_context *flb_net_dns_lookup_context_create(struct mk_event_loop *event_loop, struct flb_coro *coroutine)
+struct flb_dns_lookup_context *flb_net_dns_lookup_context_create(struct mk_event_loop *evl,
+                                                                 struct flb_coro *coroutine)
```

Comment:
[Optional] You seem to have renamed `event_loop` to `evl` to fit within the line length. Alternatively, you could format this as follows and keep the original variable name:
```c
struct flb_dns_lookup_context *flb_net_dns_lookup_context_create(
    struct mk_event_loop *event_loop, struct flb_coro *coroutine)
```